consumer/config.go

```diff
@@ -5,39 +5,61 @@ import (
 
 	"github.com/Shopify/sarama"
 	"github.com/heetch/felice/codec"
+	"gopkg.in/retry.v1"
 )
 
 // Config is used to configure the Consumer.
 type Config struct {
 	*sarama.Config
 
+	// KafkaAddrs holds kafka brokers addresses. There must be at least
```

Comment:
I tend to think that the Config should contains only things that configures how the Consumer should behaves.

Whereas Metrics and Logger are just side effect thing that don't act on the Consumer directly.

For that reason, I don't think that they should go inside the Config.

Reply:
I don't understand the distinction. Both Metrics and Logger are used by the Consumer, otherwise they wouldn't be there. So they both affect how the Consumer behaves, no?

Reply:
IMHO Config should holds only things that configure the behavior of the Consumer itself (how it consumes, where, ...) and things like Logger and Metrics are just side effect that don't have any effect in consuming.
Having them or not, the Consumer should work anyway.